blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/LocatorFetchRunnable.java

```diff
@@ -156,7 +173,21 @@ public void finishExecution(long waitStart, RollupExecutionContext executionCont
         }
     }
 
-    public int processLocator(int rollCount, RollupExecutionContext executionContext, RollupBatchWriter rollupBatchWriter, Locator locator) {
+    public int processLocator(int rollCount, RollupExecutionContext executionContext,
+                              RollupBatchWriter rollupBatchWriter, Locator locator, boolean isSlotBeingRerolled) {
+
+        if (isSlotBeingRerolled) {
+            Instrumentation.markLocatorReRolled();
+        } else {
+            Instrumentation.markLocatorRolled();
+        }
+
+        return processLocator(rollCount, executionContext, rollupBatchWriter, locator);
+    }
+
+    public int processLocator(int rollCount, RollupExecutionContext executionContext,
```

Comment:
Do we want to expose this out to caller? I.e: allowing them to skip Instrumentation? I'm thinking we may not want to. So make it protected (test classes can still call)